openmls/src/framing/errors.rs

```diff
@@ -6,94 +6,84 @@
 use crate::{
     credentials::CredentialError, error::LibraryError, tree::secret_tree::SecretTreeError,
 };
-use openmls_traits::types::CryptoError;
-use tls_codec::Error as TlsCodecError;
+use thiserror::Error;
 
-implement_error! {
-    pub enum MlsPlaintextError {
-        Simple {
-            NotAnApplicationMessage = "The MlsPlaintext message is not an application message.",
-            UnknownSender = "Sender is not part of the group",
-            InvalidSignature = "The MlsPlaintext signature is invalid",
-            InvalidMembershipTag = "The MlsPlaintext membership tag is invalid",
-        }
-        Complex {
-            LibraryError(LibraryError) = "A LibraryError occurred",
-            CodecError(TlsCodecError) = "TLS Codec error",
-            CredentialError(CredentialError) = "See [`CredentialError`](`crate::credentials::CredentialError`) for details.",
-            VerificationError(VerificationError) = "See [`VerificationError`](`VerificationError`) for details.",
-            CryptoError(CryptoError) = "See [`CryptoError`](openmls_traits::types::CryptoError) for details.",
-        }
-    }
+/// MlsCiphertext error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum MlsCiphertextError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("Couldn't find a ratcheting secret for the given sender and generation.")]
```

Comment:
From the consumer's standpoint, wouldn't it be more helpful to get something like "The given message couldn't be decrypted as it was too far out-of-order relative to previously received messages."?

Reply:
This is not a public error (despite what it says). I'll look at that if and when I expose it.